solrmonitor/solrmonitor.go

```diff
@@ -311,25 +311,41 @@ func (c *SolrMonitor) shouldWatchChildren(path string) bool {
 }
 
 func (c *SolrMonitor) dataChanged(path string, data string, version int32) error {
-	if !strings.HasPrefix(path, c.solrRoot+"/collections/") || !strings.HasSuffix(path, "/state.json") {
-		return fmt.Errorf("unknown dataChanged: %s", path)
-	}
```

Comment:
Of note. This is a semantic change. Previously this allowed monitoring `state.json` files outside of  paths with `strings.HasPrefix(path, c.solrRoot+"/collections/")`. I think the new formulation makes more sense maybe. But I think there is opportunity for simplification.

Reply:
I think this still required both, right? If either prefix or suffix didn't exist, it'd fall to the error.